### PR TITLE
quark

### PR DESCRIPTION
**Pull request by paulofaria**

### CODE REVIEW

File: Sources/Body.swift [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        case .asyncSender(let sender):
+            let drain = AsyncDrain()
+            sender(drain) { closure in
+                completion {
+                    try closure()
+                    return (.buffer(drain.data), drain.data)
+                }
+            }
+
+        default:
```
Comment (noppoMan) [inline Sources/Body.swift:153]: @paulofaria  I think we can convert all types of Body to `buffer` in `asyncBecomeBuffer` with something like the following additional cases.

``` swift
case .buffer(let data):
    completion {
        return (.buffer(data), data)
    }

case .receiver(let receiver):
    completion {
        let data = Drain(for: receiver, timingOut: deadline).data
        return (.buffer(data), data)
    }

case .sender(let sender):
    completion {
        let drain = Drain()
        try sender(drain)
        let data = drain.data
        return (.buffer(data), data)
    }
```

https://github.com/noppoMan/S4/commit/2d2d635de93570a724636a237e7bafdf8af4bb65 has a little bit my lazy...
If you took care of my future PR for about it(that talked on Slack), sorry for my meddling.

Comment (paulofaria) [inline Sources/Body.swift:153]: This is awesome!

> If you took care of my future PR for about it(that talked on Slack), sorry for my meddling.
> Sorry. I didn't understand this part.

I added you as a contributor to Zewo organization. If you want you can pull this branch (`quark`) and commit to it directly.

Comment (noppoMan) [inline Sources/Body.swift:153]: No worries. If you don't mind please adopt and merge my proposal. And I think we can get tests for `Body` from S4's commit history.

Comment (paulofaria) [inline Sources/Body.swift:153]: @noppoMan oh you man merge your PR at S4?

Comment (noppoMan) [inline Sources/Body.swift:153]: @paulofaria ohh I mean please add my additional codes for asyncBecomeBuffer that I wrote the above to this PR.(If you like it)

**Pull request closed without merging**